doc/index.rst

```diff
@@ -13,22 +13,44 @@ metrics.
 This package contains efficient Python implementations of several popular
 metric learning algorithms.
 
+Supervised Algorithms
+---------------------
+Supervised metric learning algorithms take as inputs points `X` and target
+labels `y`, and learn to make points from the same class (for classification)
+or with close target value (for regression) close to each other, and points
+from different classes or with distant target values far away from each
+other.
+
 .. toctree::
-   :caption: Algorithms
    :maxdepth: 1
 
    metric_learn.covariance
    metric_learn.lmnn
-   metric_learn.itml
-   metric_learn.sdml
-   metric_learn.lsml
    metric_learn.nca
    metric_learn.lfda
+   metric_learn.mlkr
+
+Semi-Supervised Algorithms
```

Comment:
I would rather use something like "Weakly supervised algorithms" to avoid confusion with the classic semi-supervised setting (i.e., a subset of points are labeled and other are not).

Reply:
I agree
Done